pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -127,6 +130,40 @@ func (s *startSpec) getTargetID(podStatus *kubecontainer.PodStatus) (*kubecontai
 	return &targetStatus.ID, nil
 }
 
+func calcRestartCountByLogDir(path string) (int, error) {
+	// if the path doesn't exist then it's not an error
+	if _, err := os.Stat(path); err != nil {
+		return 0, nil
+	}
+	restartCount := int(0)
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return 0, err
+	}
+	if len(files) == 0 {
+		return 0, err
+	}
+	restartCountLogFileRegex := regexp.MustCompile(`(\d+).log(\..*)?`)
```

Comment:
Is it possible for the filename to have non-`\d` characters? Why isn't the match expression for the suffix consistent with the same matching as the start of the string?

Reply:
Impossibe for the log file to have non \d characters: https://github.com/rphillips/kubernetes/blob/8c0814e915661d18a7dd179c332869c21940e598/pkg/kubelet/kuberuntime/helpers.go#L168-L170

Reply:
The suffix can optionally contain a '.rotated' or '.rotated.gz'... so those rotated files should count in the calculation.

Reply:
Ah okay we are just matching the num + `.log` + possibly any trailing characters `.whatever`. This is good, can we add one more test case for multiple logs present with `.whatever`?